CRM/Admin/Form/Setting/Search.php

```diff
@@ -102,4 +96,20 @@ public static function getContactReferenceOptions() {
     return [1 => ts('Contact Name')] + CRM_Core_OptionGroup::values('contact_reference_options', FALSE, FALSE, TRUE);
   }
 
+  /**
+   * Presave callback for contact_reference_options and contact_autocomplete_options.
+   *
+   * Ensures "1" is always contained in the array.
+   *
+   * @param $value
+   * @return bool
+   */
+  public static function enableOptionOne(&$value) {
```

Comment:
I feel like we need someone off the form layer for this - should apply to when it's set by api too

Reply:
Right, this is a callback set in the settings metadata. So prior to this PR it would only get handled correctly in the form layer. Now it will work everywhere.

Reply:
@colemanw right - but I think the location of the function is wrong - I'm not sure where it should go - maybe a new class for setting callbacks, maybe setting bao, maybe select values - but not on the form

Reply:
Yea I agree but could we take care of that in a diff PR. There are already other metadata-declared callbacks in this form class so they will all need to be relocated.

Reply:
I guess - but you are adding a callback here - still I'm OK for you to leave out of scope